src/Workspaces/Remote/ServiceHub/Services/DiagnosticAnalyzer/DiagnosticComputer.cs

```diff
@@ -585,7 +585,10 @@ private async Task<CompilationWithAnalyzersCacheEntry> CreateCompilationWithAnal
         }
 
         // Evaluate project analyzers after host analyzers to ensure duplicates in analyzerMapBuilder are
-        // overwritten with project analyzers if/when applicable.
+        // overwritten with project analyzers if/when applicable. Make sure we clear out our set so any
+        // duplicates are considered.
+        referenceSet.Clear();
```

Comment:
Right now I'm running this PR just to see if this fails any CI. I'm suspicious of this comment entirely though. It seems if you were to have the same analyzer in both the host _and_ a project, there's still a risk that they might be different instances, and this would mean you will later try passing the project analyzer to the host CompilationWithAnalyzers. I'm not sure why there aren't just two maps.

Reply:
note: i feel like we may want to address this area at a high level.  meaning, we need to fully define what it means to have the same host/project analyzer, and if we want/need to support that.  For example, jared's approach to tearing, where we just stated we run what was in the IDE only.

I would be very amenable to such an approach if it simplifies things greatly.

Reply:
Looking at this code further, that comment is indeed suspicious -- just doing the referenceSet.Clear() then means if we have two analyzers with the same identity but different instances (say they're from different ALCs) then things might go badly.